lib/replicate.js

```diff
@@ -30,6 +30,44 @@ function replicate (feed, initiator, opts) {
     if (stream.destroyed) return
     if (stream.opened(feed.key)) return
 
+    if (opts.noise !== false && opts.onfeedauthenticate) {
+      if (!stream.remotePublicKey) {
+        feed.ifAvailable.wait()
+        stream.setMaxListeners(0)
+        stream.on('close', onhandshake)
+        stream.on('handshake', onhandshake)
+        return
+      }
+      feedauthenticate()
+      return
+    }
+    replicatePeer()
+  }
+
+  function onhandshake () {
+    feed.ifAvailable.continue()
+    stream.off('close', onhandshake)
+    stream.off('handshake', onhandshake)
+    feedauthenticate()
+  }
+
+  function feedauthenticate () {
+    if (stream.destroyed) return
+    if (stream.opened(feed.key)) return
+    feed.ifAvailable.wait()
+    opts.onfeedauthenticate(feed, stream.remotePublicKey, function (err) {
```

Comment:
why is the feed passed actually? is that because this is a corestore feature in practice?

Reply:
If we just pass the discoveryKey it's a lot easier to port to the stream layer later on.

Reply:
I am passing the feed for an authentication method to have a bit more flexibility in choosing what to authenticate, such as: discoveryKey, publicKey, or size limits etc.

Reply:
How is that different than doing:

``` js
feed.replicate({
  onfeedauthenticate (remotePublicKey, done) {
    if (feed.....) 
  }
})
```

Reply:
Ok 👍 It seems a bit weird that we add a "corestore/other-many-core" feature here then, as I think that should be implemented there but we can try it out - ie it might be removed in a future major.

Reply:
I hope it won't cause trouble down the road.